fuel/streams.py

```diff
@@ -37,12 +37,36 @@ class AbstractDataStream(object):
     sources : tuple of strings
         The names of the data sources returned by this data stream, as
         given by the dataset.
+    produces_examples : bool
+        Whether this data stream produces examples (as opposed to batches
+        of examples).
 
     """
     def __init__(self, iteration_scheme=None, axis_labels=None):
         self.iteration_scheme = iteration_scheme
         self.axis_labels = axis_labels
 
+    @property
+    def produces_examples(self):
+        if self.iteration_scheme:
+            return self.iteration_scheme.requests_examples
+        elif not hasattr(self, '_produces_examples'):
+            raise ValueError("cannot infer type of stream for {} instance; "
+                             "set the produces_examples attribute to True "
+                             "(for example streams) or False (for batch "
+                             "streams).".format(self.__class__.__name__))
+        else:
+            return self._produces_examples
+
+    @produces_examples.setter
+    def produces_examples(self, value):
+        if self.iteration_scheme:
+            raise ValueError("cannot set produces_examples on {} instance; "
+                             "determined by iteration scheme {}".format(
+                                 self.__class__.__name__,
+                                 self.iteration_scheme))
+        self._produces_examples = value
+
     def get_data(self, request=None):
```

Comment:
Hmm. That's interesting.
On Jul 11, 2015 5:58 PM, "vdumoulin" <notifications@github.com> wrote:

> In fuel/streams.py
> <https://github.com/mila-udem/fuel/pull/188#discussion_r34417005>:
>
> > +            raise ValueError("cannot infer type of stream for {} instance; "
> > +                             "set the produces_examples attribute to True "
> > +                             "(for example streams) or False (for batch "
> > +                             "streams).".format(self.__class__.__name__))
> > +        else:
> > +            return self._produces_examples
> > +
> > +    @produces_examples.setter
> > +    def produces_examples(self, value):
> > +        if self.iteration_scheme:
> > +            raise ValueError("cannot set produces_examples on {} instance; "
> > +                             "determined by iteration scheme {}".format(
> > +                                 self.__class__.__name__,
> > +                                 self.iteration_scheme))
> > +        self._produces_examples = value
> > +
> >      def get_data(self, request=None):
>
> There's the Merge transformer which doesn't use get_data at all.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mila-udem/fuel/pull/188/files#r34417005>.
>